pulumi/infra/nomad_job.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional
+
+import pulumi_nomad as nomad
+from infra.config import DEPLOYMENT_NAME
+
+import pulumi
+
+
+class NomadJob(pulumi.ComponentResource):
+    def __init__(
+        self,
+        name: str,
+        vars: pulumi.Output,
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        super().__init__("grapl:NomadJob", name, None, opts)
+
+        self.grapl_core = nomad.Job(
+            resource_name=f"{DEPLOYMENT_NAME}-grapl-core-job",
+            jobspec=self._file("../../nomad/grapl-core.nomad"),
+            hcl2=nomad.JobHcl2Args(enabled=True, vars=vars),
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+
+    def _file(self, nomad_file: str) -> str:
```

Comment:
prefer `_contents` or something but whatever

Reply:
I'll split the difference and call it _file_contents so its more explicit